pkg/repo/sysmeta.go

```diff
@@ -0,0 +1,141 @@
+package repo
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/bacalhau-project/bacalhau/pkg/storage/util"
+)
+
+type SystemMetadata struct {
+	RepoVersion     int       `yaml:"Version"`
```

Comment:
Looking at the yaml tag, `Version` can be confusing as it might indicate bacalhau version. Since it represents the version of the `DataDir`, maybe we should call it `DataDirVersion`